packages/local_auth/local_auth_platform_interface/lib/local_auth_platform_interface.dart

```diff
@@ -0,0 +1,116 @@
+  /// one. Anything that is not user fixable, such as no biometric sensor on
+  /// device, will be returned as a [PlatformException].
+  ///
+  /// [stickyAuth] is used when the application goes into background for any
+  /// reason while the authentication is in progress. Due to security reasons,
+  /// the authentication has to be stopped at that time. If stickyAuth is set
+  /// to true, authentication resumes when the app is resumed. If it is set to
+  /// false (default), then as soon as app is paused a failure message is sent
+  /// back to Dart and it is up to the client app to restart authentication or
+  /// do something else.
+  ///
+  /// Provide [authStrings] if you want to
+  /// customize messages in the dialogs.
+  ///
+  /// Setting [sensitiveTransaction] to true enables platform specific
+  /// precautions. For instance, on face unlock, Android opens a confirmation
+  /// dialog after the face is recognized to make sure the user meant to unlock
+  /// their phone.
+  ///
+  /// Setting [biometricOnly] to true prevents authenticates from using non-biometric
+  /// local authentication such as pin, passcode, and passcode.
+  ///
+  /// Throws an [PlatformException] if there were technical problems with local
+  /// authentication (e.g. lack of relevant hardware). This might throw
+  /// [PlatformException] with error code [otherOperatingSystem] on the iOS
+  /// simulator.
+  Future<bool> authenticate({
+    required String localizedReason,
+    bool useErrorDialogs = true,
+    bool stickyAuth = false,
+    required Map<String, String> authStrings,
+    bool sensitiveTransaction = true,
+    bool biometricOnly = false,
+  }) async {
+    throw UnimplementedError('authenticate() has not been implemented.');
+  }
+
+  /// Returns a list of enrolled biometrics.
+  ///
+  /// Returns a [Future] List<BiometricType> with the following possibilities:
+  /// - BiometricType.face
+  /// - BiometricType.fingerprint
+  /// - BiometricType.iris (not yet implemented)
+  Future<List<BiometricType>> getAvailableBiometrics() async {
+    throw UnimplementedError(
+        'getAvailableBiometrics() has not been implemented.');
+  }
+
+  /// Returns true if device is capable of checking biometrics or is able to
+  /// fail over to device credentials.
+  ///
+  /// Returns a [Future] bool true or false:
+  Future<bool> isDeviceSupported() async {
+    throw UnimplementedError('isDeviceSupported() has not been implemented.');
+  }
+
+  /// Returns true if auth was cancelled successfully.
+  /// Returns false if there was some error or no auth in progress.
+  ///
+  /// Returns [Future] bool true or false:
+  Future<bool> stopAuthentication() async {
+    throw UnimplementedError('stopAuthentication() has not been implemented.');
+  }
+}
+
+/// The default interface implementation acting as a placeholder for
+/// the native implementation to be set.
+class DefaultLocalAuthPlatform extends LocalAuthPlatform {}
```

Comment:
This seems like a good approach for a green-field plugin, but it would effectively be a regression in `local_auth` in some edge cases (such as existing third-party implementations of the method channel for other platforms). So as much as I'd love to not have a method channel sitting around that we don't actually use, I think we should make the default implementation be the current method channel (as with other plugins here).

Reply:
Since no platform interface was published before, I'm assuming you mean forks of this plugin that make use of the previous method channel name in implementations for other platforms?

I've added a default implementation that should be 1:1 with the previous functionality. Please let me know if that is what you meant.

Reply:
> I'm assuming you mean forks of this plugin that make use of the previous method channel name in implementations for other platforms?

Not forks, just implementations in general. See https://github.com/go-flutter-desktop/plugins/tree/master/url_launcher for an example. This is an implementation of `url_launcher` for a third-party embedding; it's not a fork, and it contains no Dart code, it's just a native implementation with a method channel listener that exactly matches what `url_launcher` sends. As long as that's registered by whatever plugin mechanism the embedding uses, and as long as the method channel doesn't change, this kind of thing Just Works.

It's fragile, but before federation introduced the platform interface concept it was the only way for other embeddings to support existing plugins.

(A more likely scenario in the case of local_auth is people mocking out the method channel in tests, which prior to platform interfaces was a common way of mocking out plugins in tests.)

What you've added here is indeed what I was looking for :)